apis/python/src/tiledbsoma/_index_util.py

```diff
@@ -3,26 +3,34 @@
 SOMATileDBContext which would otherwise ensue.
 """
 
+from typing import Optional
+
 import numpy as np
 import pandas as pd
+from somacore.query.types import IndexLike
 
 from tiledbsoma import pytiledbsoma as clib
 
 from .options import SOMATileDBContext
 
 
-def build_index(
-    keys: np.typing.NDArray[np.int64], context: SOMATileDBContext
-) -> clib.IntIndexer:
+def tiledbsoma_build_index(
+    keys: np.typing.NDArray[np.int64],
+    *,
+    context: Optional[SOMATileDBContext] = None,
+    thread_count: int = 4,
+) -> IndexLike:
     """Builds an indexer object compatible with :meth:`pd.Index.get_indexer`."""
```

Comment:
nit: docstring needs to define arguments and return

Reply:
Done